src/battle-log.ts

```diff
@@ -315,6 +315,7 @@ class BattleLog {
 				node.style.display = 'none';
 				node.className = 'revealed ' + node.className;
 				count++;
+				if (count >= lineCount && lineCount !== 0) break;
```

Comment:
This is going to remove the first `lineCount ` lines; we want the last `lineCount` lines.

Reply:
You'll need to change the `for (const node of this.innerElem.childNodes as any) {` loop, but sadly it's too far up for me to suggest a change.

Reply:
This is probably best done in two loops – one to grab the matches, and one to modify the last `n` matches.

That's how your jQuery solution works, it's just that you have to do it manually here.

Reply:
...I'll write it for you.

Reply:
```ts
	hideChatFrom(userid: ID, showRevealButton = true, lineCount = 0) {
		const classStart = 'chat chatmessage-' + userid + ' ';
		let nodes = [];
		for (const node of this.innerElem.childNodes as any) {
			if (node.className && (node.className + ' ').startsWith(classStart)) {
				nodes.push(node);
			}
		}
		if (this.preemptElem) {
			for (const node of this.preemptElem.childNodes as any) {
				if (node.className && (node.className + ' ').startsWith(classStart)) {
					nodes.push(node);
				}
			}
		}
		if (lineCount) nodes = nodes.slice(-lineCount);

		for (const node of nodes) {
			if (node.className && (node.className + ' ').startsWith(classStart)) {
				node.style.display = 'none';
				node.className = 'revealed ' + node.className;
			}
		}
		if (!nodes.length || !showRevealButton) return;
		const button = document.createElement('button');
		button.name = 'toggleMessages';
		button.value = userid;
		button.className = 'subtle';
		button.innerHTML = `<small>(${nodes.length} line${nodes.length > 1 ? 's' : ''} from ${userid} hidden)</small>`;
		const lastNode = nodes[nodes.length - 1];
		lastNode.appendChild(document.createTextNode(' '));
		lastNode.appendChild(button);
	}
```

Reply:
Thank you, apologies.